src/test/java/org/minions/devfund/marylin/Squarelotron.java

```diff
@@ -0,0 +1,267 @@
+        this.numberOfRings = this.size - (this.size / 2);
+        this.squarelotron = new int[this.size][this.size];
+        for (int i = 0; i < this.size; i++) {
+            for (int j = 0; j < this.size; j++) {
+                this.squarelotron[i][j] = squarelotron[i][j];
+                this.squarelotron[i][j] = squarelotron[i][j];
+            }
+        }
+    }
+
+
+    @Override
+    public int hashCode() {
+        return Arrays.hashCode(squarelotron);
+    }
+
+    /**
+     * Compare two Squarelotron.
+     *
+     * @param object Squarelotron instance.
+     * @return Squarelotron instance.
+     */
+    @Override
+    public boolean equals(final Object object) {
+        if (!(object instanceof Squarelotron)) {
+            return false;
+        }
+
+        Squarelotron squarelotronTemp = (Squarelotron) object;
+        for (int i = 0; i < this.squarelotron.length; i++) {
+            for (int j = 0; j < this.squarelotron.length; j++) {
+                if (this.squarelotron[i][j] != squarelotronTemp.squarelotron[i][j]) {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Duplicates an array of arrays.
+     *
+     * @param original array of array.
+     * @return the duplicated array.
+     */
+    private int[][] duplicateGrid(final int[][] original) {
+        int size = original.length;
+        int[][] backupGrid = new int[size][size];
+        for (int i = 0; i < size; i++) {
+            for (int j = 0; j < size; j++) {
+                backupGrid[i][j] = original[i][j];
+            }
+        }
+        return backupGrid;
+    }
+
+    /**
+     * Prints the Squarelotron.
+     */
+    void printSquarelotron() {
+        for (int i = 0; i < squarelotron.length; i++) {
+            for (int j = 0; j < squarelotron.length; j++) {
+                System.out.print(" " + squarelotron[i][j] + " ");
+            }
+            System.out.println();
+        }
+    }
+
+    /**
+     * upsideDownFlipCompleteMatrix without check rings.
+     *
+     * @param matrix array.
+     * @return the matrix that is upside down flip completely.
+     */
+    int[][] upsideDownFlipCompleteMatrix(final int[][] matrix) {
+        int[][] resultGrid = duplicateGrid(matrix);
+        int size = matrix.length;
+        int temp = 0;
+        for (int i = size - 1; i >= size / 2; i--) {
+            for (int j = 0; j < size; j++) {
+                resultGrid[i][j] = matrix[temp][j];
+                resultGrid[temp][j] = matrix[i][j];
+            }
+            temp++;
+        }
+        return resultGrid;
+    }
+
+    /**
+     * upsideDownFlipInternalMatrix distinc to ring #1.
+     *
+     * @param matrix to be upside down flip the internal matrix.
+     * @return the matrix result-
+     */
+    int[][] upsideDownFlipInternalMatrix(final int[][] matrix) {
+        int[][] resultGrid = duplicateGrid(matrix);
+        int size = matrix.length;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "size" which hides the field declared at line 10. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)